crates/rpc/src/v03/method/estimate_message_fee.rs

```diff
@@ -0,0 +1,393 @@
+use crate::{
+    context::RpcContext,
+    v02::{method::call::FunctionCall, types::reply::FeeEstimate},
+};
+use pathfinder_common::{BlockId, EthereumAddress};
+use serde::Deserialize;
+
+use super::common::prepare_handle_and_block;
+
+#[derive(Deserialize, Debug, PartialEq, Eq)]
+pub struct EstimateMessageFeeInput {
+    message: FunctionCall,
+    sender_address: EthereumAddress,
```

Comment:
This seems to be unused, which is somewhat of a red flag to me.

In the example code in the issue `CallL1Handler.to_internal()` is used, which seems to prepend `sender_address` to the calldata.

Reply:
For reference: https://github.com/starkware-libs/cairo-lang/blob/361fe32d5930db340ea78fe05aedfe706f6c9405/src/starkware/starknet/services/api/feeder_gateway/request_objects.py#L49

Reply:
Nice catch!

I had a feeling that it was just too easy to be true :)

Reply:
Added a simple contract with trivial L1 handler for testing, it works now.